data/src/main/java/com/linkedin/data/schema/PdlBuilder.java

```diff
@@ -51,6 +52,7 @@
       "record", "typeref", "union", "null", "true", "false"
   ));
   private static final char ESCAPE_CHAR = '`';
+  private static final Pattern NON_IDENTIFIER_CHARS = Pattern.compile(".*[^0-9a-zA-Z_-].*");
```

Comment:
Tbh this is kind of confusing, it took me a minute to realize why this also contains `.*`...

Perhaps it would make more sense to use the inverse:
```
IDENTIFIER_CHARS = Pattern.compile("[0-9a-zA-Z_-]+")
```
Then do...
```
!IDENTIFIER_CHARS.matcher(propertyKey).matches()
```

Reply:
done